### PR TITLE
S-COM

### PR DESCRIPTION
**Pull request by Glloyd**
Fuck, committed to the wrong repo. Gimme a sec to push the rest of the stuff and then I'll merge this.

### CODE REVIEW

**Comment by Glloyd** [comment]
I'm running into an issue where DD crashes upon starting. It just started now and I have no idea what's causing it. Can someone else try compiling this then loading it up in DD and tell me what happens?

**Comment by scrdest** [comment]
Dammit. I'd test it, but I'm at my parents' and I don't know the router password to forward the port.

**Comment by Glloyd** [comment]
Can you get onto it and start a round? It crashes every time I join.

**Comment by scrdest** [comment]
Right now, I'm figuring if the runtimes are caused by the settings. Also, you might want to go on Steam for that :P

**Comment by Glloyd** [comment]
I've been trying to figure this thing out for the last three hours and I've got a pounding headache. I'm going to take a quick nap and come back to this with a clear head. The issue started somewhere between https://github.com/Glloyd/UristMcStation/commit/4eb3cef6c50b77c82b9ba675ef2db7538d96b728 and https://github.com/Glloyd/UristMcStation/commit/b9e791f92ddf8955ae01acba81ad5982cd5b6e8f but I'm not sure where and why, and if it's just something with my local copy. I also can't find the source of those runtimes related to the chem_grenades, which weren't happening earlier today. It's weird as fuck.

**Comment by scrdest** [comment]
It's called on an unsimulated floor, so it's likely a mapping issue. You added a map in the second one, which I suspect is the cause.

**Comment by scrdest** [comment]
OK, it seems that kind of a derpy behavior when a completely random thing is affected might be related to a compiler bug that shits out these kind of write-to-read-only runtimes when too much procs are running/too much instructions per proc, but that's been solved in 507 apparently.

**Comment by scrdest** [comment]
Jesus Harrison Christ, BYOOOOOOOOOOND!

Updating to 507.1280 fixes the issue with the runtimes. So yes, too many stuff handled by one proc. 

But the thing is, if you open BYOND, you get an update button - which updates to 506 line, which is a buggy-ass piece of crap in that regard.

...but that would be too easy. the thing is, even after updating, the DD freezes upon attempting to join.

**Comment by scrdest** [comment]
OK, I enabled logging from the 507 DD; the first time, without logging, it failed to start up at all, but this time, I've managed to run it all the way until Geometry processing before it froze.

**Comment by scrdest** [comment]
What did you do with the Glloydstation map in Glloyd@b9e791rf ('more stuff') commit? The game seems to freeze at roughly Geometry processing.

**Comment by Nienhaus** [comment]
Does loading up a different map that isn't Glloystation work?

**Comment by scrdest** [comment]
Didn't have the opportunity to check, holiday preparations.

**Comment by Nienhaus** [comment]
I tried it, nothing changed.

**Comment by scrdest** [comment]
Does it hang or crash for you?

**Comment by Nienhaus** [comment]
I get runtimes, no crash.

**Comment by scrdest** [comment]
So you can join normally? Runtimes can be fixed by compiling with the beta version.

**Comment by Nienhaus** [comment]
I was able to ghost around on the Bay map

**Comment by scrdest** [comment]
Okay, thanks to Nien, I now know that the code itself is perfectly fine. The gamemode fires and the S-Com equipping works, it didn't teleport me but it's probably because Exodus does not have the teleport locs defined.

The game freezes on Processing Geometry, which is an atmos setup process, so most likely placing the new stuff on the map causes an endless atmos loop somewhere.

**Comment by scrdest** [comment]
ERROR LOCATED AND FIXED!

It's the dumbest thing possible - the S-COM shuttles in the S-COM HQ are simulated, whereas the HQ itself isn't, and the shuttles' airlocks are only icons, not actual airlocks. Most likely, if a simulated area is airlocked, it doesn't cause issues at setup, since it's covered if it's already opening to unsimulated in-game, but if it starts open to an unsimulated area, the setup goes crazy.

After unsimulated-rwalling both shuttles, the gamemode starts up and runs perfectly fine, teleports, reassigns and all working properly. However, CVRTs throw a runtime still, after I see if airlocking the shuttle fixes the issue I'll look into that.

**Comment by Glloyd** [comment]
Whaaaaat. That's the weirdest shit. It was working with the simulated turfs and shuttles up until last night. What the fuck.

**Comment by scrdest** [comment]
I'm running a test to see if airlocking the shuttles fixes it.

**Comment by Nienhaus** [comment]
You should get on steam

**Comment by scrdest** [comment]
Setting the block_air on the adjacent turfs might work.

**Comment by scrdest** [comment]
IA! IA! BYOND FTAGHN!

**Comment by Glloyd** [comment]
fuck byond so hard.

File: code/modules/urist/gamemodes/scom/scom.dm [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+var/list/scomspawn3 = list()
+
+/datum/game_mode/scom //the joke is that 90% of this stuff is handled by other procs
+	name = "scom"
+	config_tag = "scom"
+	required_players = 0
+	required_players_secret = 18
+	votable = 0
+
+/datum/game_mode/announce() //guys, are my comments informative yet?
```
Comment (scrdest) [inline code/modules/urist/gamemodes/scom/scom.dm]: Shouldn't it be /game_mode/scom/announce? Otherwise you're overriding announce()

Comment (Glloyd) [inline code/modules/urist/gamemodes/scom/scom.dm]: Fuck.

**Pull request merged**